common/src/main/java/com/gregtechceu/gtceu/api/data/chemical/ChemicalHelper.java

```diff
@@ -164,19 +168,36 @@ public static UnificationEntry getUnificationEntry(ItemLike item) {
     public static List<ItemLike> getItems(UnificationEntry unificationEntry) {
         return UNIFICATION_ENTRY_ITEM.computeIfAbsent(unificationEntry, entry -> {
             var items = new ArrayList<ItemLike>();
-            for (TagKey<Item> tag : getTags(unificationEntry.tagPrefix, unificationEntry.material)) {
-                for (Holder<Item> itemHolder : Registry.ITEM.getTagOrEmpty(tag)) {
-                    items.add(itemHolder.value());
+            for (TagKey<Item> tag : getTags(entry.tagPrefix, entry.material)) {
+                Item unified = AlmostUnifiedAdapter.getPreferredItemForTag(tag);
+                if (unified != null) {
+                    items.add(unified);
+                } else {
+                    for (Holder<Item> itemHolder : Registry.ITEM.getTagOrEmpty(tag)) {
+                        items.add(itemHolder.value());
+                    }
                 }
```

Comment:
Suggestion for less nesting (optional):

```suggestion
                if (unified != null) {
                    items.add(unified);
                    continue;
                }
                
                for (Holder<Item> itemHolder : Registry.ITEM.getTagOrEmpty(tag)) {
                    items.add(itemHolder.value());
                }
```